### PR TITLE
Remove deprecated function calls to hiera and friends

### PR DESCRIPTION
**Pull request by mickenordin**
Switch hiera('blablabla') -> lookup('blablabla', undef, undef, undef)
Switch hiera('blablabla', 'NOT_SET_IN_HIERA') -> safe_hiera('blablabla')
Switch hiera('blablabla', 'blabla') -> lookup('blablabla', undef, undef, 'blabla')
Switch hiera_array('blablabla', ['blabla']) -> lookup('blablabla', undef, undef, ['blabla'])
Switch safe_hiera('blablabla', 'NOT_SET_IN_HIERA') -> safe_hiera('blablabla')
Switch safe_hiera('blablabla', 'blabla') -> lookup('blablabla', undef, undef, 'blabla')

Note: If types are known at compiletime we also add that like:

String $str = lookup('str', String, undef, '')

Also switch the safe_hiera function over to lookup function in ruby code

More info on this here: https://www.puppet.com/docs/puppet/5.5/hiera_automatic.html#looking-up-data-with-hiera

Note: do not squash, there are lint commits here as well

### CODE REVIEW

File: manifests/backuphost.pp [hunk 1]
```diff
@@ -3,7 +3,7 @@
   String        $chroot,
   String        $mountpoint,
   String        $user   = 'backup',
-  Array[String] $allow_clients = hiera('backup_pfx'),
+  Array[String] $allow_clients = lookup('backup_pfx', Array[String], undef, ['130.242.125.68', '130.242.121.73']),
```
Comment (mickenordin) [inline manifests/backuphost.pp:6]: I don't know if this is controversial or not, I don't want to lock out anyone from ssh, so set hoppjerka and joppherka here as default

**Review by mickenordin — COMMENTED** [review]
Should be safe, one possible controversial change is commented on below

File: manifests/security/configure_sshd.pp [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # @param port            The port to configure sshd to listen on. If undef, we leave it to sshd to decide.
 class sunet::security::configure_sshd (
   Boolean $configure_sftp = true,
-  Optional[Integer] $port = hiera('sunet_ssh_daemon_port', undef),
+  Optional[Integer] $port = lookup('sunet_ssh_daemon_port', Optional[integer], undef,  undef),
```
Comment (theseal) [inline manifests/security/configure_sshd.pp:6]: Is this a valid syntax?

File: manifests/snippets/secret_file.pp [hunk 3]
```diff
@@ -13,7 +13,7 @@
     default  => $path
   }
   $safe_key = regsubst($hiera_key, '[^0-9A-Za-z_]', '_', 'G')
-  $data = hiera($hiera_key, hiera($safe_key))
+  $data = lookup($hiera_key, undef, undef, lookup($safe_key, undef, undef, undef))
```
Comment (theseal) [inline manifests/snippets/secret_file.pp:16]: Inception?

**Review by theseal — APPROVED** [review]
Please have a look at my comments and make sure that syntax is correct.
Besides that I see no issue merging. Would have done it my self if it wasn't my last Friday before vacation. Merge with care and caution 🙂

File: manifests/snippets/secret_file.pp (hunk 3 above)
Comment (mickenordin) [reply manifests/snippets/secret_file.pp:16]: You think it should be brokendown into two lines instead?

Comment (theseal) [reply manifests/snippets/secret_file.pp:16]: Two lines or maybe two variables but that might be what you ment? Easier to read and understand.

File: manifests/security/configure_sshd.pp (hunk 2 above)
Comment (mickenordin) [reply manifests/security/configure_sshd.pp:6]: Yes. As far as I can tell, and as my testing can conclude.

https://www.puppet.com/docs/puppet/5.5/hiera_automatic.html#puppet_lookup

 lookup( <NAME>, [<VALUE TYPE>], [<MERGE BEHAVIOR>], [<DEFAULT VALUE>] ) 
<VALUE TYPE> (data Type) - A data type that must match the retrieved value; if not, the lookup (and catalog compilation) will fail. Defaults to Data which accepts any normal value.

**Pull request merged**